pkg/controllers/dashboard/fleetcharts/controller.go

```diff
@@ -38,6 +40,15 @@ func Register(ctx context.Context, wContext *wrangler.Context) error {
 	}
 
 	wContext.Mgmt.Setting().OnChange(ctx, "fleet-install", h.onSetting)
+	// watch cluster repo `rancher-charts` and enqueue the setting to make sure the latest fleet is installed after catalog refresh
```

Comment:
This makes sure that if rancher-charts has changed, we immediately enqueue the setting which triggers fleet chart to to be installed. This makes sure fleet always get the latest version as soon as chart refresh happens.